lib/diaspora_federation/entity.rb

```diff
@@ -264,21 +315,52 @@ def simple_node(doc, name, value)
       end
     end
 
-    # @param [Nokogiri::XML::Element] root_node xml nodes
+    def json_data
+      enriched_properties.map {|key, value|
+        type = self.class.class_props[key]
+
+        if type.instance_of?(Class) && value.respond_to?(:to_json_hash)
+          entity_data = value.to_json_hash[:entity_data]
+          [key, entity_data] unless entity_data.nil?
+        else
+          [key, value]
+        end
```

Comment:
Well, there is also a `to_json` method from `JSON::Ext::Generator::GeneratorMethods::Hash/Array` modules, which are independent of Rails. If one doesn't use ActiveSupport here, he can fallback to these methods. However I have no idea of how to test it with our test suite. If we remove ActiveSupport, we can use these replacements as well, so having this call is probably doesn't make us dependent on ActiveSupport.